packages/react-sdk/src/utils/warn-once.ts

```diff
@@ -0,0 +1,15 @@
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+let warnOnce = (_: string) => {};
```

Comment:
`eslint-disable` can be avoided if you replace `{}` with `undefined`

Reply:
Copypasted from nextjs, I think (not sure) that empty body is explicitly set here, so smart build can fully remove all calls to this function in production.

Reply:
For example esbuild
```
function a(_: string) {}
const c = a('hello');
```
`yarn esbuild --minify --tree-shaking=true --loader=ts < file.js`

minify this to empty code


BUT 

```
const a = (_: string) => {};
a('hello')
```

is not minified.

I don't know does above minified or not.

Reply:
2 thoughts:
1. this would be great to have everywhere
2. maybe we could even overwrite console.warn to log only once?

Reply:
is there an npm package that does exactly this? so we can use it in designer and everywhere else?

Reply:
cc @TrySound, @rpominov  
the idea of all such helpers that they and all calls into must be eliminated at production, even if it's not wrapped like in my code with `if (process.env.NODE_ENV) { warnOnce`

In the past all such helpers came with babel etc transforms. But nowadays we have rollup, esbuild, webpack and this feature is tool-dependent. Probably no npm with support for everything exists, BTW my knowledge is not enough.

Reply:
in jss and many other places we use https://github.com/alexreardon/tiny-warning/blob/master/src/index.js which warns in non-production env only and works everywhere well, but it doesn't do the warnOnce logic, so we kinda need the same but for warnOnce

Reply:
> maybe we could even overwrite console.warn to log only once?

Please don't. I can easily imagine spending a day questioning my sanity because logs don't work :)

Reply:
Here is the exact package you need https://github.com/satya164/warn-once/blob/main/index.js

Reply:
And I know the guy, he is great

Reply:
> Please don't. I can easily imagine spending a day questioning my sanity because logs don't work :)

yep, dumb idea

Reply:
btw needs to be checked

Reply:
it shouldn't, that's not how it works in jss with tiny-warning at least, @TrySound was helping there to implement it, I wouldn't worry about it right now, please go ahead use that warnOnce package, if something is off we can still fix rollup for this later